packages/terra-date-time-picker/src/DateTimeUtils.js

```diff
@@ -87,12 +87,16 @@ class DateTimeUtils {
     const newDate = momentDate.clone();
     const timeFormat = hasSeconds ? 'HH:mm:ss' : 'HH:mm';
     const date = moment(time, timeFormat, true);
+    const timeZone = momentDate.tz() || moment.tz.guess();
 
-    if (hasSeconds) {
-      return newDate.hour(date.get('hour')).minute(date.get('minute')).second(date.get('second'));
+    let dateTimeString;
+    if (date.isValid()) {
+      dateTimeString = newDate.format('YYYY-MM-DD').concat(' ').concat(date.format(timeFormat));
+    } else {
+      dateTimeString = newDate.format('YYYY-MM-DD');
```

Comment:
Since this method is for updating the time in the provided `momentDate`, if the provided `time` is invalid (in this case, `date.isValid()`) would be false), I think we should just exist and return the original `momentDate` instead of proceeding since it would update it with the current time.

Reply:
Yeah that works too, updated here 52065b6ece74801302990a047817302ab5c59aee